ui/app/routes/vault/cluster/recovery/snapshots.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { service } from '@ember/service';
+import { hash } from 'rsvp';
+
+import { SystemListStorageRaftSnapshotLoadListEnum } from '@hashicorp/vault-client-typescript';
+
+import type ApiService from 'vault/services/api';
+import type Capabilities from 'vault/services/capabilities';
+
+export default class RecoverySnapshotsRoute extends Route {
+  @service declare readonly api: ApiService;
+  @service declare readonly capabilities: Capabilities;
+
+  async model() {
+    const path = 'sys/storage/raft/snapshot/snapshot-load';
+    const capabilities = await this.capabilities.fetch([path]);
+
+    const canLoadSnapshot = capabilities[path]?.canUpdate;
+
+    try {
+      const { keys } = await this.api.sys.systemListStorageRaftSnapshotLoad(
+        SystemListStorageRaftSnapshotLoadListEnum.TRUE
+      );
+
+      const snapshots = await Promise.all(
+        (keys ?? []).map((key: string) => this.api.sys.systemReadStorageRaftSnapshotLoadId(key))
+      );
+
+      return hash({
+        snapshots,
+        canLoadSnapshot,
+      });
+    } catch (e) {
```

Comment:
We only want to catch if the error is a 404, if the person doesn't have permission to list snapshots then we want to surface that error